birdset/datamodule/beans_datamodule.py

```diff
@@ -104,4 +106,29 @@ def label_to_id_fn(batch):
             ),
         )
         
-        return dataset
\ No newline at end of file
+        return dataset
+    
+    
+    def _preprocess_data(self, dataset):
+        """
+        Preprocess the data. This calls the _ksamples function to select k samples per class and calls the embedding extraction function. If multilabel is the task we will one hot encode. 
+        """
+
+        # Check if actually a dict
+        dataset = self._ksamples(dataset)
+        dataset = self._compute_embeddings(dataset)
+
+        if self.dataset_config.task == 'multilabel':
+            log.info(">> One-hot-encode classes")
+            dataset = dataset.map(
+                self._classes_one_hot,
+                batched=True,
+                batch_size=500,
+                load_from_cache_file=True,
+                num_proc=self.dataset_config.n_workers,
+            )
+
+        return dataset
```

Comment:
@XgamerTV This code is not specific to BEANS, and should be put into `EmbeddingsDatamodule` to easily integrate it into other Datasets (e.g. BirdSet)